trin-core/src/portalnet/overlay_service.rs

```diff
@@ -1327,11 +1420,129 @@ mod tests {
+                assert_eq!(ConnectionDirection::Outgoing, status.direction);
+            }
+            _ => panic!(),
+        };
+
+        // Check ping queue for first ENR.
+        // Key for node should be present.
+        assert!(service.peers_to_ping.contains_key(&enr1.node_id()));
+
+        // Check ping queue for second ENR.
+        // Key for node should be present.
+        assert!(service.peers_to_ping.contains_key(&enr2.node_id()));
+    }
+
+    #[tokio::test]
+    async fn process_discovered_enrs_known_enrs() {
+        let mut service = task::spawn(build_service());
+
+        // Generate random ENRs to simulate.
+        let (sk1, mut enr1) = generate_random_remote_enr();
+        let (_, enr2) = generate_random_remote_enr();
+
+        let key1 = kbucket::Key::from(enr1.node_id());
+        let key2 = kbucket::Key::from(enr2.node_id());
+
+        let status = NodeStatus {
+            state: ConnectionState::Connected,
+            direction: ConnectionDirection::Outgoing,
+        };
+        let data_radius = U256::from(u64::MAX);
+
+        let node1 = Node {
+            enr: enr1.clone(),
+            data_radius: data_radius.clone(),
+        };
+        let node2 = Node {
+            enr: enr2.clone(),
+            data_radius: data_radius.clone(),
+        };
+
+        // Insert nodes into routing table.
+        let _ = service
+            .kbuckets
+            .write()
+            .insert_or_update(&key1, node1, status);
+        assert!(std::matches!(
+            service.kbuckets.write().entry(&key1),
+            kbucket::Entry::Present { .. }
+        ));
+        let _ = service
+            .kbuckets
+            .write()
+            .insert_or_update(&key2, node2, status);
+        assert!(std::matches!(
+            service.kbuckets.write().entry(&key2),
+            kbucket::Entry::Present { .. }
+        ));
+
+        // Modify first ENR to increment sequence number.
+        let updated_udp: u16 = 9000;
+        let _ = enr1.set_udp(updated_udp, &sk1);
+        assert_ne!(1, enr1.seq());
+
+        let mut enrs: Vec<Enr> = vec![];
+        enrs.push(enr1.clone());
+        enrs.push(enr2.clone());
+        service.process_discovered_enrs(enrs);
+
+        // Check routing table for first ENR.
+        // Node should be present with ENR sequence number equal to 2.
+        match service.kbuckets.write().entry(&key1) {
+            kbucket::Entry::Present(entry, _status) => {
+                assert_eq!(2, entry.value().enr.seq());
+            }
+            _ => panic!(),
+        };
+
+        // Check routing table for second ENR.
+        // Node should be present with ENR sequence number equal to 1.
+        match service.kbuckets.write().entry(&key2) {
+            kbucket::Entry::Present(entry, _status) => {
+                assert_eq!(1, entry.value().enr.seq());
+            }
+            _ => panic!(),
+        };
+    }
+
```

Comment:
Maybe it is worth also checking the use case when we process our own local `Enr` and ignoring it.